swarm/api/api.go

```diff
@@ -204,52 +200,55 @@ func (m *MultiResolver) getResolveValidator(name string) ([]ResolveValidator, er
 	return rs, nil
 }
 
+// SetNameHash - on a MultiResolver
 func (m *MultiResolver) SetNameHash(nameHash func(string) common.Hash) {
 	m.nameHash = nameHash
 }
 
 /*
-Api implements webserver/file system related content storage and retrieval
+API implements webserver/file system related content storage and retrieval
 on top of the dpa
 it is the public interface of the dpa which is included in the ethereum stack
 */
-type Api struct {
+type API struct {
 	resource *mru.Handler
 	dpa      *storage.DPA
 	dns      Resolver
 }
 
-//the api constructor initialises
-func NewApi(dpa *storage.DPA, dns Resolver, resourceHandler *mru.Handler) (self *Api) {
-	self = &Api{
+// NewAPI - the api constructor initialises
+func NewAPI(dpa *storage.DPA, dns Resolver, resourceHandler *mru.Handler) (self *API) {
+	self = &API{
 		dpa:      dpa,
 		dns:      dns,
 		resource: resourceHandler,
 	}
 	return
 }
 
-// to be used only in TEST
-func (self *Api) Upload(uploadDir, index string, toEncrypt bool) (hash string, err error) {
-	fs := NewFileSystem(self)
+// Upload - to be used only in TEST
```

Comment:
We should extract test-only symbols to separate files with build tags on them.

Reply:
I think this is outside the scope of this issue / PR. But we can create a new issue to take care of these. Would you like me to do that?

Reply:
Let's discuss it in roundtable next week maybe. It's no rush. BUt yes, out of scope.